### PR TITLE
CTAA-1409 - Fixed randomly generated id on handshakes

### PR DESCRIPTION
**Pull request by dusanjencik**
## Changes
 
Fixed and algorithm for random generating id for handshake. Now it is able to use also numbers from 0 - 999.
 
## Solved issues
 
- [Issue #1409](https://tasks.pxp-x.com/browse/CTAA-1409)

### CODE REVIEW

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/HandshakeCodewordRepository.kt [hunk 1]
```diff
@@ -57,4 +58,18 @@ class HandshakeCodewordRepositoryImpl(
     private fun getResourceId(identifier: String): Int {
         return contextInteractor.resources.getIdentifier("handshake_code_$identifier", "string", contextInteractor.packageName)
     }
+
+    override fun zeroPrefixed(number: Int): String {
+        val value = abs(number)
+        if (value < RANDOM_IDENTIFICATION_MIN ||
+            value > RANDOM_IDENTIFICATION_MAX ||
+            number < 0) {
+            Timber.e(SilentError("Invalid input number $number"))
```
Comment (JU5T1C3) [inline app/src/main/java/at/roteskreuz/stopcorona/model/repositories/HandshakeCodewordRepository.kt:67]: what are you doing if the number is incorrect? you still go on with it, right?

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/HandshakeCodewordRepository.kt [hunk 2]
```diff
@@ -15,39 +17,38 @@ interface HandshakeCodewordRepository {
...
-                    return identification
-                }
-            }
-        } catch (exc: NumberFormatException) {
-            Timber.e(SilentError(exc))
-            return identification
-        }
-
-        val parts = listOf(identification.substring(0, 2), identification.substring(2))
+    override val identificationNumber: Int = Random.nextInt(from = RANDOM_IDENTIFICATION_MIN, until = RANDOM_IDENTIFICATION_MAX)
```
Comment (dusanjencik) [inline app/src/main/java/at/roteskreuz/stopcorona/model/repositories/HandshakeCodewordRepository.kt:47]: nextInt second value is exclusive. plus 1 to it.

Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/model/repositories/HandshakeCodewordRepository.kt]: Updated

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/HandshakeCodewordRepository.kt (hunk 1 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/model/repositories/HandshakeCodewordRepository.kt:67]: yes, what do you suggest to do instead?
`If there is number greater than 9999, the result is last 4 characters of the string value.`

**Pull request merged**